specification/resources/resource-manager/Microsoft.Resources/stable/2022-12-01/subscriptions.json

```diff
@@ -0,0 +1,947 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {}
+      }
+    },
+    "/subscriptions/{subscriptionId}/locations": {
+      "get": {
+        "tags": [
+          "Subscriptions"
+        ],
+        "operationId": "Subscriptions_ListLocations",
+        "summary": "Gets all available geo-locations.",
+        "description": "This operation provides all the locations that are available for resource providers; however, each resource provider may support a subset of this list.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "includeExtendedLocations",
+            "in": "query",
+            "required": false,
+            "type": "boolean",
+            "description": "Whether to include extended locations.",
+            "x-ms-parameter-location": "method"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of locations.",
+            "schema": {
+              "$ref": "#/definitions/LocationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "GetLocationsWithASubscriptionId": {
+            "$ref": "./examples/GetLocations.json"
+          },
+          "GetLocationsWithExtendedLocations": {
+            "$ref": "./examples/GetLocationsWithExtendedLocations.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}": {
+      "get": {
+        "tags": [
+          "Subscriptions"
+        ],
+        "operationId": "Subscriptions_Get",
+        "description": "Gets details about a specified subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
```

Comment:
@gauravkumar1491 

Would you let us know why previous local `SubscriptionIdParameter` with `x-ms-parameter-location=method` get replaced in 2022-12-01 by `common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter`.

The change from `x-ms-parameter-location=method` to `x-ms-parameter-location=client` breaks SDK and user.

Reply:
Also impact Go SDK. @gauravkumar1491 could you please confirm?

Reply:
@weidongxu-microsoft and @tadelesh The PR was flagged with Staging LintDiff errors where referencing the common types was required. These are common-types and should always be referenced instead of creating local parameters. Got approval in the Breaking Changes team meeting as well

Reply:
We can ignore LintDiff (I would be glad to, if this is the only error).

All your RP is about subscriptions, hence it is actually not make sense to have "subscriptionId" in client parameter.

Before the change, user can call `client.get(subscriptionId)` to get information of any subscription they has interest on.

After the change, user can only call `client.get()` for the info of their subscription configured during client initialization.

Reply:
If you agree to revert back to prior SubscriptionIdParameter with `x-ms-parameter-location=method`, I can talk to Jeffrey and Mike on whether they can approve the revert PR.

Reply:
@weidongxu-microsoft  I can add to x-ms-parameter-location=method to[ "../../../../../common-types/resource-management/v5/types.json#/parameters](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json)/SubscriptionIdParameter"

Reply:
No, you cannot. It is a shared model. If you change it there, every other RP that uses it will get broken (and most RP does use a single subscriptionId).

Reply:
I've added a PR to revert https://github.com/Azure/azure-rest-api-specs/pull/24088